### ISSUE TITLE
Suggestion: Helion to pick the payment at the card picking time

### ISSUE THREAD

**Issue by VanderLeal** [bafolts/terraforming-mars#2498]
Another suggestion from someone that plays with people in different time zones and it affects the waiting time considerably.
The fact Helion need to choose how to pay at THE END of the card picking phase makes it slower as requiring an extra action from someone out of order.

Say player Helion picks his card. Player Tharsis is the 1st player, and also pick his card. Now Tharsis needs to wait Helion to come back just so he can say how much heat he wants to use to pay for his cards. Waiting for a player to play his turn is fine. But additional waiting just for something that could've been solved when picking, seems unnecessary, if it can be avoided (of course). 

For now, our group is even removing Helion just because of that.

Cheers from Down Under. Love your work guys.

PS: Sorry for the typo at the title

**Comment by ChrisTheCat** [comment]
If there's some reason this suggestion isn't possible, please at least indicate the Helion player is on-turn when they are being asked to make this decision. Currently, it incorrectly looks like the first player is on turn.

**Comment by ChrisTheCat** [comment]
Despite the lack of activity on the issue, this is still something that needs to be fixed.

**Comment by bafolts** [comment]
We should be able to implement this by adding a function to `DeferredQueue` to allow us to run the deferred actions by `Player`. If when a player is done with their research phase we see the `DeferredQueue` has something queue for a `Player` we can run those actions. Only once that queue is empty is that player then done with their research phase.
